File: src/main/scala/mod/iceandshadow3/compat/block/BMateria.java
package mod.iceandshadow3.compat.block;

import mod.iceandshadow3.basics.block.IMateria;
import net.minecraft.block.SoundType;
import net.minecraft.block.material.Material;

public abstract class BMateria implements IMateria {
	final Material mcmat;
	final SoundType snd;

	protected BMateria(Material material, SoundType sndtype) {
		mcmat = material;
		snd = sndtype;
	}
	
	@Override
	public int getBaseOpacity() {
		return mcmat.isOpaque() ? 0xff : 0;
	}
}

File: src/main/scala/mod/iceandshadow3/lib/compat/block/BMateria.scala
package mod.iceandshadow3.lib.compat.block

import mod.iceandshadow3.lib.block.IMateria
import net.minecraft.block.SoundType
import net.minecraft.block.material.Material

abstract class BMateria(protected[compat] val mcmat: Material, protected[compat] val sound: SoundType)
extends IMateria {
	override def getBaseOpacity = if (mcmat.isOpaque) 0xff else 0
}